webapp/src/components/AssetPage/SaleActionBox/BuyNFTButtons/BuyWithCryptoButton/BuyWithCryptoButton.tsx

```diff
@@ -1,11 +1,29 @@
-import { Network } from '@dcl/schemas'
+import { useCallback } from 'react'
+import { NFTCategory, Network } from '@dcl/schemas'
 import { t } from 'decentraland-dapps/dist/modules/translation'
+import { getAnalytics } from 'decentraland-dapps/dist/modules/analytics'
 import { Button, Mana } from 'decentraland-ui'
+import * as events from '../../../../../utils/events'
+import { isNFT } from '../../../../../modules/asset/utils'
 import { Props } from './BuyWithCryptoButton.types'
 
 export const BuyWithCryptoButton = (props: Props) => {
+  const { asset, onClick, ...rest } = props
+
+  const handleOnClick = useCallback(() => {
```

Comment:
What if we mimic the click procedure so we can keep the `onClick` handler that comes from the Button component?
```suggestion
  const handleOnClick = useCallback((e) => {
```